cr-examples/onnx/src/main/java/oracle/code/onnx/OnnxInterpreter.java

```diff
@@ -1,13 +1,18 @@
 package oracle.code.onnx;
 
 import oracle.code.onnx.ir.OnnxOp;
+import oracle.code.onnx.ir.OnnxOps.*;
 
 import java.util.List;
 
 public class OnnxInterpreter {
     public static Object interpret(Class<? extends OnnxOp> opClass,
                                    List<Object> inputs,
                                    List<Object> attributes) {
+        if (opClass == Add.class) {
+            // @@@ type detection
```

Comment:
Each `OnnxOp` declares a static final field `SCHEMA` with all the information to drive reflective operation, so i believe it should be possible to eventually generalize the execution in the implementation of this method. The class initializer of `OnnxOps` can build an onnx op function from class to schema for easier schema access.

Although it's not very efficient i believe the best eventual approach for the interpreter is to generate an ONNX code model using the op schema, convert the model to protobuf, then execute in its own session. That will exercise all the functionality.